source/server/listener_manager_impl.cc

```diff
@@ -1014,6 +1015,13 @@ Network::DrainableFilterChainSharedPtr ListenerFilterChainFactoryBuilder::buildF
 
   auto& config_factory = Config::Utility::getAndCheckFactory<
       Server::Configuration::DownstreamTransportSocketConfigFactory>(transport_socket);
+  if (transport_socket.name() == Extensions::TransportSockets::TransportSocketNames::get().Quic &&
```

Comment:
This doesn't seem right; in general we don't assign any meaning to the `name` of extensions in Envoy in xDS v3, we use the type URL to determine what type of extension it is. `name` is just basically an opaque label.

Reply:
Changed to check ListenerConfig object instead. If it has udp listener factory and the factory isTransportConnectionless() returns true, assume it is using QUIC.